### PR TITLE
Migrate from self-navigator to @polymer/app-route

### PR DESCRIPTION
**Pull request by lukebjerring**
## Description
Rewrites the main `interop` and `wpt-results` components as "views", which are switched between using virtual navigation (drastically reducing load time when switching).

In addition, it drops our custom `self-navigator.js` code in favour of Polymer's `app-route` which handles virtual navigation better.

## Review Information
- Visit /admin/flags, enable the feature, visit homepage, change query, navigate back/forward etc.

### CODE REVIEW

**Comment by Hexcles** [comment]
This looks very promising and will likely also fix #985 . Luke, could you take a look at the the broken tests?

**Comment by Hexcles** [comment]
@lukebjerring awesome! Can we also remove a bunch of templates and just leave `index.html`?

**Comment by Hexcles** [comment]
Still seeing many test failures

**Comment by Hexcles** [comment]
I fixed the build in #1303 , but WebDriver tests related to `wpt-results` and `wpt-interop` are still failing.

File: Makefile [hunk 1]
```diff
@@ -151,6 +151,9 @@ firefox_install: firefox_deps bzip2 wget java
 firefox_deps:
 	sudo apt-get install -qqy --no-install-suggests $$(apt-cache depends firefox | grep Depends | sed "s/.*ends:\ //" | tr '\n' ' ')
 
+geckodriver: node-selenium-standalone
+	./webapp/node_modules/.bin/selenium-standalone install --singleDriverInstall=firefox
```
Comment (Hexcles) [inline Makefile]: Hmm, I thought this was installed in the `post-install` script of `selenium-standalone`, no?

If we need to do this, we should:
```suggestion
	cd webapp; `npm bin`/selenium-standalone install --singleDriverInstall=firefox
```

File: webapp/components/test-runs-query.js [hunk 2]
```diff
@@ -296,26 +350,40 @@ const TestRunsUIQuery = (superClass, opt_queryCompute) => class extends TestRuns
...
 };
-// TODO(lukebjerring): Support to & from in the builder.
-const testRunsUIQueryComputer =
-  'computeTestRunUIQueryParams(shas, aligned, master, labels, productSpecs, to, from, maxCount, offset, diff, search, pr, runIds)';
 
 TestRunsQuery.Computer = testRunsQueryComputer;
 TestRunsUIQuery.Computer = testRunsUIQueryComputer;
 
 export { TestRunsQuery, TestRunsUIQuery };
+
```
Comment (Hexcles) [inline webapp/components/test-runs-query.js]:
```suggestion
```

**Comment by lukebjerring** [comment]
@Hexcles  - I don't see the bad request you mention; have you got a repro link?

**Comment by Hexcles** [comment]
(I'll merge it when I get back on Tuesday.)

**Comment by Hexcles** [comment]
Finally got some time to sort out the tests and merge conflicts. Today looks like a good day to merge it :)

**Pull request merged**